Index layout (#74)

Mikiyas-hs opened the pull request:
## What does this change?

Resolves issue #74 

For this user story, I created the homepage layout where visitors can easily explore the main sections of the site. I made sure that all the components could come together as a whole and put them in a grid layout so that the end user can look at a well-structured homepage layout, so that they can easily navigate and understand what the site offers.

https://github.com/user-attachments/assets/85cd168f-2765-482a-b7e2-85fb556deb89


## How Has This Been Tested?
<!-- Link to test results in the Wiki-->

- [ ] [User test]()
- [x] [Accessibility test]()
- [x] [Performance test]()
- [x] [Responsive Design test]()
- [ ] [Device test]()
- [x] [Browser test]()

## Images
<img width="1896" height="868" alt="Schermafbeelding 2025-11-05 130114" src="https://github.com/user-attachments/assets/3698cc4a-ca39-402a-bc06-81e80d67331e" />
<img width="1897" height="865" alt="Schermafbeelding 2025-11-05 130104" src="https://github.com/user-attachments/assets/5ef69bf6-c235-4478-848d-e8d02da788ec" />
<img width="1892" height="861" alt="Schermafbeelding 2025-11-05 130029" src="https://github.com/user-attachments/assets/6ac30a61-30c2-47a7-a15f-a586a96bdb56" />



## How to review

How to review
If you open this branch and start the localhost, you’ll be able to see my components as soon as you click the link.
I would like your feedback on my code:

Am I following proper coding conventions?
Is it clear and understandable?
I would like your feedback on the design:

Does anything look off?
Could anything be improved?
Did I forget anything?

vsheo committed:
chore: project structure fix #44 #74

fixed our routes from issue 44
added fetch url so Mikiyas can continue with the homepage layout

vsheo committed:
Merge branch 'dev' into index-layout

Mikiyas-hs committed:
Fix: Correct sponsor data (#74)

FIxed the data reference and image paths so sponsor logos load correctly

Mikiyas-hs committed:
Style: adjust sponsors section height and background color (#74)

Mikiyas-hs committed:
Feat: load both projects and sponsors data in layout (#74)

fetches projects and sponsors together using promise.all for faster and cleaner data loading

Mikiyas-hs committed:
Chore: prepare layout for upcoming header component (#74)

Mikiyas-hs committed:
chore: disable page server load function (#74)

Commented out the page load function since data now comes from layout

Mikiyas-hs committed:
Style: restructure page layout and add spacing (#74)

Updated the page structure with grid layout for projects and added margin below the about section for better spacing

vsheo commented on src/routes/+page.svelte:
```diff
@@ -1,8 +1,45 @@
+
+
+
+</main>
```
you can rewrite this to
```HTML
<main>
	<ContainerProjectCard {data}/>
	<About />
	<SponsorCarousel {data} />
</main>
```

vsheo commented on src/routes/+page.svelte:
```diff
@@ -1,8 +1,45 @@
+    .about-us {
+  margin-bottom: 2rem; /* of 60px, pas aan naar smaak */
+}
+    
```
this can be moved to the about component, inside the first section (CSS selector)

vsheo commented on src/lib/components/SponsorCarousel.svelte:
```diff
@@ -1,6 +1,6 @@
-    export let data
-    const sponsors = data.Sponsor
+    export let data 
+    const sponsors = data
```
this you can change to 
```JS
    const sponsors = data.sponsors
```

vsheo commented on src/routes/+page.svelte:
```diff
@@ -1,8 +1,45 @@
 <script>
-    import { SponsorCarousel, ContainerProjectCard } from '$lib'
+  // componenten importeren zet na SponsorCarousel een "," dan kan je alles erin doen
+    import { ContainerProjectCard, About, FooterInfo, ProjectCard, SponsorCarousel  } from '$lib'
```
using ContainerProjectcard will load in the ProjectCard component aswel.
ProjectCard, can be removed.

vsheo reviewed (changes requested):
with these changes the page.svelte will look cleaner

Mikiyas-hs committed:
Merge branch 'dev' into index-layout

Mikiyas-hs committed:
Merge branch 'dev' into index-layout

Mikiyas-hs committed:
style: add margin-bottom to section for spacing (#74) (#86)

Mikiyas-hs committed:
fix: reference sponsors array directly from data object (#74) (#86)

fix: correctly acces sponsors from data object

Mikiyas-hs committed:
refactor: replace logo with FooterInfo component in layout (#74) (#86)

Replaced logo with FooterInfo for a more consistent layout structure

Mikiyas-hs committed:
fix: re-enable load function to fetch sponsor data (#74) (#86)

Uncommented and restored the async load function to properly return sponsor data from the api

Mikiyas-hs committed:
refactor: simplify main layout and disable unused project grid styling (#74) (#86)

Removed unused components and commented out .projects-grid styling for cleaner layout structure

vsheo reviewed (approved):
the layout looks great now
